webpack/ForemanTasks/Components/TasksTable/TasksTable.js

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect } from 'react';
+import { Spinner } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import { Table } from 'foremanReact/components/common/table';
+import { STATUS } from 'foremanReact/constants';
+import MessageBox from 'foremanReact/components/common/MessageBox';
+import { translate as __ } from 'foremanReact/common/I18n';
+import Pagination from 'foremanReact/components/Pagination/PaginationWrapper';
+import createTasksTableSchema from './TasksTableSchema';
+import { updateURlQuery } from './TasksTableHelpers';
+import './TasksTable.scss';
+
+const TasksTable = ({
+  getTableItems,
+  error,
+  status,
+  results,
+  query,
+  itemCount,
+  pagination,
+  setSort,
+  setPagination,
+  sort,
+}) => {
+  useEffect(() => {
+    getTableItems();
+  }, [
+    query.state,
+    query.result,
+    query.mode,
+    query.time,
+    pagination.page,
+    pagination.perPage,
+    sort.by,
+    sort.order,
+  ]);
+  const onPageChange = ({ page, perPage }) => {
+    updateURlQuery({
+      page,
+      per_page: perPage,
+    });
+    setPagination({ page, perPage });
+  };
+
+  if (status === STATUS.ERROR) {
+    return (
+      <MessageBox
+        key="tasks-table-error"
+        icontype="error-circle-o"
+        msg={__(`Could not receive data: ${error && error.message}`)}
```

Comment:
I didn't see this props are actually used in the MessageBox component,
which currently looks like: `export const MessageBox = () => <div className="message-box-root" />;`

the check for `error` and `error.message` could be moved to the upper `if` statement.

Reply:
oops, I saw now that the `MessageBox` is being mocked :rofl: